### PR TITLE
Fix some doc references

### PR DESCRIPTION
**Pull request by gspencergoog**
This fixes a few of the many, many dartdoc warnings due to invalid references.

Sadly, fixing these is not easily automate-able.

### CODE REVIEW

File: packages/flutter/lib/src/cupertino/text_field.dart [hunk 1]
```diff
@@ -294,10 +294,6 @@ class CupertinoTextField extends StatefulWidget {
   /// Whitespace characters (e.g. newline, space, tab) are included in the
   /// character count.
   ///
-  /// If [maxLengthEnforced] is set to false, then more than [maxLength]
```
Comment (goderbauer) [inline packages/flutter/lib/src/cupertino/text_field.dart]: Why did this get removed?

**Review by goderbauer — APPROVED** [review]
LGTM

**Comment by goderbauer** [comment]
I wonder if it would be possible to catch invalid dart doc references with an analyzer lint to avoid introducing them into the code base in the first place.

File: packages/flutter/lib/src/cupertino/text_field.dart (hunk 1 above)
Comment (gspencergoog) [reply packages/flutter/lib/src/cupertino/text_field.dart]: Because that applies to the Material text field, but not Cupertino.

**Comment by gspencergoog** [comment]
Every one of this was an invalid reference for IntelliJ already, there just isn't any visual representation that it's an invalid reference.  So the analyzer already knows at some level.

**Comment by gspencergoog** [comment]
Added https://github.com/flutter/flutter-intellij/issues/3075 to help with this problem.

**Pull request merged**